src/utils/FlowUtils.ts

```diff
@@ -74,13 +138,23 @@ export function validateFillForDeposit(fill: Fill, deposit?: Deposit, fillFields
     return false;
   }
 
-  // Note: this short circuits when a key is found where the comparison doesn't match.
-  // TODO: if we turn on "strict" in the tsconfig, the elements of FILL_DEPOSIT_COMPARISON_KEYS will be automatically
-  // validated against the fields in Fill and Deposit, generating an error if there is a discrepency.
-  return FILL_DEPOSIT_COMPARISON_KEYS.every((key) => {
-    if (fillFieldsToIgnore.includes(key)) {
-      return true;
-    }
-    return fill[key] !== undefined && fill[key].toString() === deposit[key]?.toString();
-  });
+  if (isV2Deposit(deposit) && isV2Fill(fill)) {
+    return V2_DEPOSIT_COMPARISON_KEYS.every((key) => {
+      if (fillFieldsToIgnore.includes(key)) {
+        return true;
+      }
+      return fill[key] !== undefined && fill[key].toString() === deposit[key]?.toString();
+    });
+  }
+
+  if (isV3Deposit(deposit) && isV3Fill(fill)) {
+    return V3_DEPOSIT_COMPARISON_KEYS.every((key) => {
+      if (fillFieldsToIgnore.includes(key)) {
```

Comment:
We will need to add logic (in this PR or a follow up) to account for when `outputToken = 0x0` on the `Deposit`, then the `outputToken` must be the canonical destination token on the `Fill` and the `outputToken` will technically not match but will still be valid.

Reply:
However, we also should be able to use this function to validate SlowFillRequests with Deposits in which case the slow fill request CAN specify `outputToken = 0x0` so the `outputTokens` will match in this case

Reply:
I've updated the SpokePool so that it overwrites outputToken 0x0 immediately on ingestion. This is basically doing that at the lowest possible level and should save everyone else from having to worry about it :+1:.

Reply:
- A slow fill is valid iff `deposit.outputToken = requestSlowFill.outputToken` && `outputToken = <canonical destination token for deposit.inputToken> OR `deposit.inputToken = 0x0` and `fill.outputToken = <canonical destination token for deposit.inputToken>`.
- A fast fill is valid iff `deposit.outputToken == fill.outputToken` OR `deposit.inputToken = 0x0` and `fill.outputToken = <canonical destination token for deposit.inputToken>`

Reply:
Per https://github.com/across-protocol/sdk-v2/pull/501#discussion_r1464828125, the SpokePoolClient subs in the right outputToken in the case that the deposit set it to 0x0 :+1:.